wandb/sdk/wandb_run.py

```diff
@@ -706,6 +706,12 @@ def _init(
                 "step": self._settings.fork_from.value,
             }
 
+        if self._settings.resume_from is not None:
```

Comment:
maybe make the setting oneof? so we guranteed we don't have both fork_from and resume_from?

Reply:
or i guess now looking at the type, maybe just a filed that says what type of xx_from?

Reply:
see convo here - https://weightsandbiases.slack.com/archives/C0108SLEP6K/p1717518143092499?thread_ts=1717466352.551879&cid=C0108SLEP6K

i implemented some logic in the message deserializer that keeps the message structure the same (for backcompat purposes and separation of concerns) and ensures that we never instantiate a settings object in the backend that has both of these specified